engine/src/test/java/io/camunda/zeebe/engine/perf/JobBackoffRestoreMigrationPerformanceTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.perf;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbCompositeKey;
+import io.camunda.zeebe.db.impl.DbForeignKey;
+import io.camunda.zeebe.db.impl.DbLong;
+import io.camunda.zeebe.db.impl.DbNil;
+import io.camunda.zeebe.engine.state.instance.JobRecordValue;
+import io.camunda.zeebe.engine.state.instance.JobStateValue;
+import io.camunda.zeebe.engine.state.mutable.MutableJobState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessingState;
+import io.camunda.zeebe.engine.util.ProcessingStateExtension;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.test.util.jmh.JMHTestCase;
+import io.camunda.zeebe.test.util.junit.JMHTest;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ExtendWith(ProcessingStateExtension.class)
+@Warmup(iterations = 100, time = 1)
+@Measurement(iterations = 50, time = 1)
+@Fork(
+    value = 1,
+    jvmArgs = {"-Xmx4g", "-Xms4g"})
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(org.openjdk.jmh.annotations.Scope.Benchmark)
+public class JobBackoffRestoreMigrationPerformanceTest {
+
+  public static final Logger LOG =
+      LoggerFactory.getLogger(JobBackoffRestoreMigrationPerformanceTest.class.getName());
+
+  public MutableJobState jobState;
+  private ZeebeDb zeebeDb;
+  private MutableProcessingState processingState;
+  private TransactionContext transactionContext;
+  private long count;
+
+  @Benchmark
+  public long measureMigrationTime(final JobBackoffRestoreMigrationPerformanceTest state) {
```

Comment:
I don;t think that this parameter will be set ? 🤔 I think you need to use a field here

Reply:
https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/perf/EngineLargeStatePerformanceTest.java

Reply:
I tried different setups for the test, but with the same result 😄 But I'm sure that I'm missing something, don't know what

Reply:
You need to create the state more manually like the db and state classes, as I did in the linked benchmark class. Here we use the TestEngine, which is to create the engine. 

You need something similar, I\m not sure whether it will work with the extension.